### PR TITLE
MiddlewareGet ASIS Fix for OMPI_Snapshot.py

### LINKED ISSUE #677: OMPI_Snapshot Plugin only works with ASIS enabled
The ompi_snapshot plugin fails when ASIS is not enabled with the following error:
```
2018-07-24 17:46:53.452164 LOGGING results for MiddlewareGet:OMPIMaster
2018-07-24 17:46:53.452185 =======================================
2018-07-24 17:46:53.452197 Exception was raised: <type 'exceptions.UnboundLocalError'> local variable 'status' referenced before assignment
2018-07-24 17:46:53.452207 =======================================
2018-07-24 17:46:53.452461 Traceback (most recent call last):

  File "/home/pgottesm/PyMTT/mtt/pylib/Tools/Executor/sequential.py", line 283, in execute
    plugin.execute(stageLog, keyvals, testDef)

  File "/home/pgottesm/PyMTT/mtt/pylib/Tools/Fetch/OMPI_Snapshot.py", line 211, in execute
    log['status'] = status

UnboundLocalError: local variable 'status' referenced before assignment
```
From a brief look over the code it looks like the log values are mostly set directly and the variable `status` is only set in one case, when ASIS is enabled. In all other cases, MTT will crash.

### PR DESCRIPTION
**Pull request by DebRez**
Fixes Error: local variable 'status' referenced before assignment. If
as-is is not chosen it now correctly clears out the directory, loads
a new tarball, and creates a version file if requested.

Fixes #677

Signed-off-by: Deb Rezanka <rezanka@lanl.gov>

### CODE REVIEW

File: pylib/Tools/Fetch/OMPI_Snapshot.py [hunk 1]
```diff
@@ -168,16 +169,29 @@ def execute(self, log, keyvals, testDef):
...
+                    log['status'] = 1
+                    log['stderr'] = "download for tarball " + tarball_name + "url: " + download_url + "FAILED"
+                    return
+                # untar the tarball
+                testDef.logger.verbose_print("untarring tarball " + tarball_name)
+                status, stdout, stderr, _ = testDef.execmd.execute(None, ["tar", "-zxf", tarball_name], testDef)
+                if 0 != status:
+                    log['status'] = 1
+                    log['stderr'] = "untar of tarball " + tarball_name + "FAILED"
+                    return
```
Comment (PeterGottesman) [inline pylib/Tools/Fetch/OMPI_Snapshot.py]: Is there a way to remove this code duplication?

**Comment by PeterGottesman** [comment]
This fixes the issue, thanks Deb!

File: pylib/Tools/Fetch/OMPI_Snapshot.py (hunk 1 above)
Comment (DebRez) [reply pylib/Tools/Fetch/OMPI_Snapshot.py]: I tried to pull it out as a independent method but kept getting errors. Finally gave up on it.

**Comment by DebRez** [comment]
Can I go ahead and merge this? @rhc54

**Comment by rhc54** [comment]
Sure - go ahead!

**Pull request merged**